force-app/main/default/classes/FlowControlLoopsCollections.cls

```diff
@@ -1,368 +1,466 @@
-     * @param employeeName The name of the employee.
-     * @return The salary of the employee, or null if the employee is not found in the Map.
-     */
-    public static Integer getSalary(String employeeName) {
-        // Initialize a Map
-
-        // Add key-value pairs to the Map
-        // add John Doe => 50000
-        // add Jane Smith => 60000
-        // add Sam Brown => 55000
-        // add Alice Johnson => 65000
-
-        // Get the salary of the employee
-
-        // Return the salary of the employee, or null if the employee is not found in the Map
-        return null; // Replace null with the variable you used to store the result
+    return totalDollarAmt; // Replace null with the variable you used to store the result
+  }
+
+  /**
+   * Question 18
+   * Create a map that has five key-value pairs to the map, where the keys are the names of fruits and the values are their quantities.
+   * Example: addItemsToMap() should return "Apples => 5, Oranges => 10, Bananas => 15, Pears => 20, Grapes => 25"
+   * @return The Map of fruits and their quantities.
+   */
+  public static Map<String, Integer> addItemsToMap() {
+    // Initialize an empty Map
+    Map<String, Integer> fruitQuantity = new Map<String, Integer>();
+
+    // Add key-value pairs to the Map
+    // add Apples => 5
+    // add Oranges => 10
+    // add Bananas => 15
+    // add Pears => 20
+    // add Grapes => 25
+    fruitQuantity.put('Apples', 5);
+    fruitQuantity.put('Oranges', 10);
+    fruitQuantity.put('Bananas', 15);
+    fruitQuantity.put('Pears', 20);
+    fruitQuantity.put('Grapes', 25);
+
+    return fruitQuantity; // Replace null with the variable you used to store the result
+  }
+
+  /**
+   * Question 19
+   * Create a map where the keys are the names of employees and the values are their salaries.
+   * Then returns the salary of the employee whose name is specified.
+   * Example: getSalary("John Doe") should return 50000
+   * Resource: https://developer.salesforce.com/docs/atlas.en-us.apexref.meta/apexref/apex_methods_system_map.htm#apex_System_Map_get
+   * @param employeeName The name of the employee.
+   * @return The salary of the employee, or null if the employee is not found in the Map.
+   */
+  public static Integer getSalary(String employeeName) {
+    // Initialize a Map
+    Map<String, Integer> empSalary = new Map<String, Integer>{
+      'John Doe' => 50000,
+      'Jane Smith' => 60000,
+      'Sam Brown' => 55000,
+      'Alice Johnson' => 65000
+    };
+
+    // Return the salary of the employee whose name is specified.
+    // If the employee is not found, return null
+
+    // Add key-value pairs to the Map
+    // add John Doe => 50000
+    // add Jane Smith => 60000
+    // add Sam Brown => 55000
+    // add Alice Johnson => 65000
+
+    // Get the salary of the employee
+    if (empSalary.containsKey(employeeName)) {
```

Comment:
You actually dont need this check here, if the index is not found, it will return null. You can just do `return empSalary.get(employeeName); `and it will handle the null.